### PR TITLE
[TT-11185, DX-1104] Update documentation for 5.3 (gateway)

### PR DESCRIPTION
**Pull request by buger**
## **User description**
Triggered by: titpetric

Included:

Tyk Gateway: true
Tyk Dashboard: false
Tyk MDCB false
Tyk Pump false

Intended for: 5.3
Changes sourced from: release-5.3
Config info generator branch: main

Note: TT-11185, gateway changes only (branch suffix: docs-gateway)

JIRA: https://tyktech.atlassian.net/browse/TT-11185

___

## **Type**
documentation, enhancement

___

## **Description**
- Corrected environment variable prefixes from 'EV' to 'ENV' in the gateway configuration documentation for consistency and accuracy.
- Added new environment variables for Redis TLS configuration in the gateway configuration documentation.
- Introduced environment variables for analytics and cache storage TLS configuration in the gateway configuration documentation.
- Updated the version of Tyk Gateway API from 5.2.3 to 5.3.0 in the Swagger documentation.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>gateway-config.md</strong><dd><code>Correct Environment Variable Prefixes and Add TLS Config Variables</code></dd></summary>
<hr>

tyk-docs/content/shared/gateway-config.md
<li>Corrected environment variable prefixes from 'EV' to 'ENV' for <br>consistency and accuracy.<br> <li> Added new environment variables for Redis TLS configuration.<br> <li> Introduced environment variables for analytics and cache storage TLS <br>configuration.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4188/files#diff-0a53845669feaeb0d62d4b17d24bfee5fc23482ec7562bd52e23db03373f14d0">+411/-290</a></td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>gateway-swagger.yml</strong><dd><code>Update Tyk Gateway API Version in Swagger Documentation</code>&nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/assets/others/gateway-swagger.yml
- Updated the version of Tyk Gateway API from 5.2.3 to 5.3.0.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4188/files#diff-86641d5f43d842a51680c692c0ca3cd7d7e67776df7bf1409cf97f83a885ad99">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: tyk-docs/content/shared/x-tyk-gateway.md [hunk 1]
```diff
@@ -173,171 +219,198 @@ It is necessary when the data lives within this string-encoded JSON object.
...
+be considered as 0s/empty.
 
-Tyk classic API definition: `domain`.
 
+Tyk classic API definition: `global_rate_limit.per`.
 
 ### **ListenPath**
 
+ListenPath is the base path on Tyk to which requests for this API
+should be sent. Tyk listens out for any requests coming into the host at
```
Comment (dcs3spp) [inline tyk-docs/content/shared/x-tyk-gateway.md:374]:
```suggestion
should be sent. Tyk listens for any requests coming into the host at
```

File: tyk-docs/content/shared/x-tyk-gateway.md [hunk 2]
```diff
@@ -354,323 +427,255 @@ Tyk classic API definition: `base_identity_provided_by`.
...
+Config contains configuration related to custom authentication plugin.
 
-**Field: `cache` ([Cache](#cache))**
-Cache contains the configurations related to caching.
+Tyk classic API definition: `custom_middleware.auth_check`.
 
-Tyk classic API definition: `cache_options`.
+### **SecuritySchemes**
 
+SecuritySchemes holds security scheme values, filled with Import().
```
Comment (dcs3spp) [inline tyk-docs/content/shared/x-tyk-gateway.md:670]: Checking to understand what this means. Does this mean that security schemes are populated from Go package imports somehow?

File: tyk-docs/content/shared/x-tyk-gateway.md [hunk 3]
```diff
@@ -687,302 +692,592 @@ Bundle configures custom plugin bundles.
...
-**Field: `timeout` (`int`)**
+**Field: `timeout` (`int64`)**
 Timeout is the TTL for a cached object in seconds.
 
+
 Tyk classic API definition: `cache_options.cache_timeout`.
 
 **Field: `cacheAllSafeRequests` (`boolean`)**
-CacheAllSafeRequests caches responses to (`GET`, `HEAD`, `OPTIONS`) requests overrides per-path cache settings in versions, applies across versions.
+CacheAllSafeRequests caches responses to (`GET`, `HEAD`, `OPTIONS`) requests overrides per-path cache settings in versions,
```
Comment (dcs3spp) [inline tyk-docs/content/shared/x-tyk-gateway.md:813]:
```suggestion
CacheAllSafeRequests caches responses to (`GET`, `HEAD`, `OPTIONS`) requests overrides per-path cache settings in versions. Applies across versions.
```

File: tyk-docs/content/shared/x-tyk-gateway.md [hunk 4]
```diff
@@ -687,302 +692,592 @@ Bundle configures custom plugin bundles.
...
+applies across versions.
```
Comment (dcs3spp) [inline tyk-docs/content/shared/x-tyk-gateway.md:814]: remove this after applying above suggestion

File: tyk-docs/content/shared/x-tyk-gateway.md [hunk 5]
```diff
@@ -687,302 +692,592 @@ Bundle configures custom plugin bundles.
...
+**Field: `regexpMatchIndex` (`int`)**
+RegexpMatchIndex is the index from which ID to be extracted after a match.
+Default value is 0, ie if regexpMatchIndex is not provided ID is matched from index 0.
+
+**Field: `xPathExp` (`string`)**
+XPathExp is the xpath expression to match ID.
+
+### **OAuthProvider**
+
+**Field: `jwt` ([JWTValidation](#jwtvalidation))**
```
Comment (dcs3spp) [inline tyk-docs/content/shared/x-tyk-gateway.md:1036]: Missing description for jwt and introspection

File: tyk-docs/content/shared/x-tyk-gateway.md [hunk 6]
```diff
@@ -687,302 +692,592 @@ Bundle configures custom plugin bundles.
...
+**Field: `introspection` ([Introspection](#introspection))**
+
+
+### **JWTValidation**
+
+**Field: `enabled` (`boolean`)**
+Enabled activates OAuth access token validation by introspection to a third party.
+
+**Field: `signingMethod` (`string`)**
+SigningMethod to verify signing method used in jwt - allowed values HMAC/RSA/ECDSA.
```
Comment (dcs3spp) [inline tyk-docs/content/shared/x-tyk-gateway.md:1048]:
```suggestion
SigningMethod to verify signing method used in jwt-allowed values HMAC/RSA/ECDSA.
```

File: tyk-docs/content/shared/x-tyk-gateway.md [hunk 7]
```diff
@@ -687,302 +692,592 @@ Bundle configures custom plugin bundles.
...
+**Field: `enabled` (`boolean`)**
+Enabled activates the caching mechanism for introspection responses.
+
+**Field: `timeout` (`int64`)**
+Timeout is the duration in seconds of how long the cached value stays.
+For introspection caching, it is suggested to use a short interval.
+
+### **ExternalOAuth**
+
+**Field: `enabled` (`boolean`)**
```
Comment (dcs3spp) [inline tyk-docs/content/shared/x-tyk-gateway.md:1100]: Missing field descriptions

File: tyk-docs/content/shared/x-tyk-gateway.md [hunk 8]
```diff
@@ -993,37 +1288,483 @@ ContentType is the default HTTP response body type that the gateway reads from t
...
+Internal holds the endpoint configuration, configuring the endpoint for internal requests.
+Tyk classic API definition: `version_data.versions...extended_paths.internal[*]`.
 
 **Field: `enabled` (`boolean`)**
-Enabled enables virtual endpoint.
+Enabled if set to true makes the endpoint available only for internal requests.
 
-**Field: `name` (`string`)**
-Name is the name of js function.
+### **JWT**
```
Comment (dcs3spp) [inline tyk-docs/content/shared/x-tyk-gateway.md:1299]: Missing field descriptions

File: tyk-docs/content/shared/x-tyk-gateway.md [hunk 9]
```diff
@@ -993,37 +1288,483 @@ ContentType is the default HTTP response body type that the gateway reads from t
...
+
+Notifications holds configuration for updates to keys.
+
+**Field: `sharedSecret` (`string`)**
+SharedSecret is the shared secret used in the notification request.
+
+**Field: `onKeyChangeUrl` (`string`)**
+OnKeyChangeURL is the URL a request will be triggered against.
+
+### **OAuth**
```
Comment (dcs3spp) [inline tyk-docs/content/shared/x-tyk-gateway.md:1373]: Missing field descriptions

File: tyk-docs/content/shared/x-tyk-gateway.md [hunk 10]
```diff
@@ -993,37 +1288,483 @@ ContentType is the default HTTP response body type that the gateway reads from t
...
+**Field: `trackEndpoint` ([TrackEndpoint](#trackendpoint))**
+TrackEndpoint contains the configuration for enabling analytics and logs.
+
+**Field: `doNotTrackEndpoint` ([TrackEndpoint](#trackendpoint))**
+DoNotTrackEndpoint contains the configuration for disabling analytics and logs.
+
+**Field: `requestSizeLimit` ([RequestSizeLimit](#requestsizelimit))**
+RequestSizeLimit limits the maximum allowed size of the request body in bytes.
+
+### **Path**
```
Comment (dcs3spp) [inline tyk-docs/content/shared/x-tyk-gateway.md:1458]: Missing field descriptions

File: tyk-docs/content/shared/x-tyk-gateway.md [hunk 11]
```diff
@@ -993,37 +1288,483 @@ ContentType is the default HTTP response body type that the gateway reads from t
...
+Enabled activates the Request Size Limit functionality.
+
+**Field: `value` (`int64`)**
+Value is the maximum allowed size of the request body in bytes.
+
+### **SecurityScheme**
+
+SecurityScheme defines an Importer interface for security schemes.
+
+### **Signature**
```
Comment (dcs3spp) [inline tyk-docs/content/shared/x-tyk-gateway.md:1531]: Missing field descriptions

File: tyk-docs/content/shared/x-tyk-gateway.md [hunk 12]
```diff
@@ -993,37 +1288,483 @@ ContentType is the default HTTP response body type that the gateway reads from t
...
+
+**Field: `enabled` (`boolean`)**
+Enabled activates Method Transform for the given path and method.
+
+**Field: `toMethod` (`string`)**
+ToMethod is the http method value to which the method of an incoming request will be transformed.
+
+### **TykExtensionConfigParams**
+
+TykExtensionConfigParams holds the essential configuration required for the Tyk Extension schema.
```
Comment (dcs3spp) [inline tyk-docs/content/shared/x-tyk-gateway.md:1618]: Missing descriptions

**Review by dcs3spp — CHANGES_REQUESTED** [review]
@titpetric Thanks for raising separate PR for GW. I've had a look through and review summary is:
- Missing field descriptions for:
    - OAuthProvider
    - External OAUth
    - JWT
    - OAuth
    - Path
    - Signature
    - TykExtensionConfigParams
    
  - Some line breaks introduced in descriptions/explanations. is there anyway of preventing the line breaks?
  
  - Minor grammar suggestions.
 - Peer review required before merge and release

**Review by andyo-tyk — APPROVED** [review]
Whilst there are some empty fields, I think we can publish this and then correct those in a future update.

**Review by dcs3spp — APPROVED** [review]

**Comment by dcs3spp** [comment]
Merging. Comments unresolved to make it easier to track fixes requested in future patch as discussed in https://tyktech.atlassian.net/browse/TT-11665

**Pull request merged**